drivers/sensor/bmm150/bmm150.c

```diff
@@ -663,10 +670,15 @@ static int bmm150_init(const struct device *dev)
 		return err;
 	}
 
+#if defined(CONFIG_PM_DEVICE)
```

Comment:
please, avoid usage of `CONFIG_PM_DEVICE`. If you read its Kconfig help you'll realize it's not the right choice. Please take a look to PM_DEVICE_RUNTIME.

Reply:
This KConfig option is the one that decides whether `pm_device_driver_init` is compiled or not. Why is it the wrong one then?
`pm_device_driver_init` has support for PM_DEVICE_RUNTIME stuff as well.

Reply:
PM_DEVICE is about managing devices when system goes to idle, something Nordic SoCs do not support because CONFIG_PM is not even selectable.

Reply:
This can (and needs to be) removed when using `pm_device_driver_init` properly.

Reply:
done